### PR TITLE
Rename subscription yaml files

### PR DESCRIPTION
**Pull request by nanobash**
## Summary

[Preview](https://rebilly-website--rem-git-rem-01hghkm6gr4vmzwegqx-426d10.preview.redocly.app/catalog/all/Orders/GetSubscriptionCollection/)

- [x] Rename `Subscription.yaml` to `SubscriptionOrOneTimeSale.yaml`
- [x] Rename `OneTimeOrder.yaml` to `OneTimeSale.yaml`
- [x] Rename `StorefrontOneTimeOrder.yaml` to `StorefrontOneTimeSale.yaml`
- [x] Rename `SubscriptionOrder.yaml` to `Subscription.yaml`
- [x] Rename `StorefrontSubscriptionOrder.yaml` to `StorefrontSubscription.yaml`

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by vykhrystiuk — DISMISSED** [review]

**Review by creocoder — DISMISSED** [review]

**Comment by 2e3s** [comment]
Wouldn't it break SDK generation?

File: openapi/components/schemas/Subscription.yaml [hunk 1]
```diff
@@ -1,10 +1,472 @@
...
+    format: date-time
+  rebillNumber:
+    description: Current billing period number.
+    type:
+      - 'integer'
+      - 'null'
+    readOnly: true
+    x-sortable: true
+  lineItems:
+    description: Subscription line items which queue until the next renewal (or interim) invoice is issued for the subscription.
```
Comment (noon-dawg) [inline openapi/components/schemas/Subscription.yaml:183]: suggestion: for readability
```suggestion
    description: Subscription line items which queue until the next renewal, or interim, invoice is issued for the subscription.
```

File: openapi/components/schemas/Subscription.yaml [hunk 2]
```diff
@@ -1,10 +1,472 @@
...
+    readOnly: true
+    type: array
+    deprecated: true
+    items:
+      type: object
+      description: Invoice line item. Use `isInterim` property of upcoming invoice items instead.
```
Comment (noon-dawg) [inline openapi/components/schemas/Subscription.yaml:189]: suggestion: for readability
```suggestion
      description: Invoice line item. Use the `isInterim` property of upcoming invoice items instead.
```

File: openapi/components/schemas/Subscription.yaml [hunk 3]
```diff
@@ -1,10 +1,472 @@
...
+        quantity:
+          description: Quantity of the line item.
+          type: integer
+          example: 1
+        periodStartTime:
+          description: Date and time when the period begins for this item.
+          type: string
+          format: date-time
+        periodEndTime:
+          description: Date and time when the period ends for this item.
```
Comment (noon-dawg) [inline openapi/components/schemas/Subscription.yaml:218]: suggestion: if think we should add service here (if it does not change the meaning)
```suggestion
          description: Date and time when the service period begins for this item.
          type: string
          format: date-time
        periodEndTime:
          description: Date and time when the service period ends for this item.
```

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

File: openapi/components/schemas/SubscriptionOrOneTimeSale.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type: object
+description: Subscription details.
+discriminator:
+  propertyName: orderType
+  mapping:
+    subscription-order: ./Subscription.yaml
+    one-time-order: ./OneTimeSale.yaml
+anyOf:
```
Comment (justnero) [inline openapi/components/schemas/SubscriptionOrOneTimeSale.yaml:8]: Schemas referenced here are mutually exclusive, as they don't intersect in terms of a required property `orderType` definition.
```suggestion
oneOf:
```

**Review by justnero — APPROVED** [review]

**Review by vykhrystiuk — APPROVED** [review]

**Review by creocoder — APPROVED** [review]

**Comment by nanobash** [comment]
> Wouldn't it break SDK generation/compatibility?

It's just renaming files and should not change much, but just interface names. Clarified with @justnero and should be fine to merge. Thanks for the question.

**Review by noon-dawg — APPROVED** [review]

**Pull request merged**